web/scenes/Portal/common/DeleteTeamDialog/index.tsx

```diff
@@ -56,15 +57,17 @@ export const DeleteTeamDialog = (props: DeleteTeamDialogProps) => {
 
   const onClose = useCallback(() => {
     reset();
+    setDeleteFinished(false);
     props.onClose(false);
-  }, [props, reset]);
+  }, [props, reset, setDeleteFinished]);
 
   const [deleteTeam] = useDeleteTeamMutation({
```

Comment:
can't we just use the loading in here instead of the setDeleteFinished

Reply:
@andy-t-wang I handled it like this because we updating the cache in `useEffect` on the `data` object change. I had to do this because apollo's `onCompleted` function executes only when we fetch query first time and it's not triggered when some mutation refetching `meQuery`. 

Also `meQuery` is just a react hook that uses apollo query inside, and it returns a bit modified `user` after the `useEffect`. So we need to wait not only apollo query refetch, but also the modified that is returned from the hook. 
 Otherwise we receiving not updated user in the submit and memberships array is still having already deleted item. 
 
 So, with this refactor we are submitting team delete, waiting while meQuery hook return actual data and then making correct redirect.